### PR TITLE
docs: add contributing guidelines

### PR DESCRIPTION
**Pull request by PeterMosmans**
Add initial version of contributing guidelines. Note that an opinionated list of formatters and linters is being prescribed. This will diminish bike-shedding, and will make it easier to review diffs / pull requests.

Issue: https://github.com/radicallyopensecurity/pentext/issues/13

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Thanks for taking the time to contribute!
+
+## Pull Requests
+
+PenText uses the [GitHub flow](https://guides.github.com/introduction/flow/) as
+main versioning workflow. In short,
+
+1. Fork the PenText repository
+2. Create a new branch for each feature, fix, or improvement
+3. Send a pull request from each feature branch to the **master** branch
```
Comment (syrosh) [inline CONTRIBUTING.md:12]: may we change the master branch to main?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Please create pull requests that are as atomic as possible, and only implement
+one feature, fix, or improvement at a time. That makes it easier to review and
+merge the changes.
+
+## Style Guide
```
Comment (syrosh) [inline CONTRIBUTING.md:18]: what are your thoughts on adding pre-commit or pre-push hooks to validate / auto-fix these things?

File: CONTRIBUTING.md (hunk 1 above)
Comment (PeterMosmans) [reply CONTRIBUTING.md:12]: I fully agree that each new *repository* should have its main branch named, well,  main. However, renaming an existing branch might lead to backwards compatibility issues. I see no _pressing_ need myself to rename the current structure. It's just, erm, legacy :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (PeterMosmans) [reply CONTRIBUTING.md:18]: Locally I use pre-commit and commit-msg hooks. I'm not sure if pre-push hooks can be implemented easily on GitHub on a repository. By ensuring that we validate PRs, we can kinda play for 'hook' ;)

**Pull request merged**